colcon_notification/event_handler/status.py

```diff
@@ -103,14 +105,19 @@ def __call__(self, event):  # noqa: D102
             line = data.line
             if isinstance(line, bytes):
                 line = line.decode(errors='replace')
-            match = self._progress_pattern.match(line)
-            if not match:
+            match_make = self._progress_pattern_make.match(line)
+            match_ninja = self._progress_pattern_ninja.match(line)
+            if not match_make and not match_ninja:
                 return
             job = event[1]
             progress = self._running[job].get('progress', [])
             while len(progress) > 1:
                 progress.pop()
-            progress.append(match.group(1).lstrip() + '%')
+            if match_make:
+                progress.append(match_make.group(1).lstrip() + '%')
+            if match_ninja:
+                progress.append('{:.0%}'.format(
+                    int(match_ninja.group(1))/int(match_ninja.group(2))))
```

Comment:
In the presence of both patterns (unlikely, but still), we should prefer the existing/more common pattern used by GNU make, which is not how this code will behave. Unless there is a specific reason to do the opposite, of course.